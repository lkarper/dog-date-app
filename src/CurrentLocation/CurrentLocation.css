.CurrentLocation__alert-div {
    min-height: 85px;
}

.CurrentLocation__alert-div span {
    display: inline-block;
}

.CurrentLocation__alert-div span:first-child {
    font-weight: bold;
}
