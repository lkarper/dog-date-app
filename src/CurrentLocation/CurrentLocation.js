import React from 'react';
import PropTypes from 'prop-types';
import './CurrentLocation.css';

const CurrentLocation = (props) => {
    
    const { coordinates } = props;

    return (
        <div 
            className='CurrentLocation__alert-div'
            role='alert'
        >
            <p id='current-set-coordinates'>
                {
                    coordinates.lat !== 0 || coordinates.lon !== 0
                        ? 
                            <>
                                <span>Coordinates set to:{' '}</span>
                                <br /> 
                                <span>Lat: {coordinates.lat}</span>
                                <br />
                                <span>Lon: {coordinates.lon}</span>
                            </>
                        : 
                            'No coordinates set (coordinates are optional).'
                }
            </p>
        </div>
    );
}

CurrentLocation.defaultProps = {
    coordinates: {
        lat: 0,
        lon: 0,
    },
};

CurrentLocation.propTypes = {
    coordinates: PropTypes.shape({
        lat: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
        lon: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
    }).isRequired,
};

export default CurrentLocation;
