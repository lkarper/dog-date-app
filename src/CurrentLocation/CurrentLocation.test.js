import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import CurrentLocation from './CurrentLocation';

describe('CurrentLocation component', () => {
    it('renders the UI without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<CurrentLocation />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(<CurrentLocation />);
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
