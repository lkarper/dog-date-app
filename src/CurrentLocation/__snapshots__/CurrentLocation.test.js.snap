// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentLocation component renders the UI as expected 1`] = `
<CurrentLocation
  coordinates={
    Object {
      "lat": 0,
      "lon": 0,
    }
  }
>
  <div
    className="CurrentLocation__alert-div"
    role="alert"
  >
    <p
      id="current-set-coordinates"
    >
      No coordinates set (coordinates are optional).
    </p>
  </div>
</CurrentLocation>
`;
